### PR TITLE
chore(cicd): New workflow to create/update main branch with master changes

### LINKED ISSUE #29840: Create a `main` branch that mimics the `master` branch
### Parent Issue
#29554 

### User Story
As a developer, I want to create a `main` branch that mimics the `master` branch so that we can gradually migrate to using `main` as the default branch, ensuring a smooth transition with automatic updates.

### Acceptance Criteria
* A `main` branch is created automatically by a GitHub action and is an exact copy of the `master` branch, it includes the same restrictions and protection rules.
* Whenever a change is merged into `master`, the `main` branch is updated to match the latest commit in `master`.
* The GitHub action to keep `main` updated runs automatically after every merge into `master`.

### Proposed Objective
Implement a GitHub action that creates the `main` branch and keeps it in sync with the `master` branch after every merge into `master`.

### Sub-Tasks & Estimates
- [ ] Research and create the GitHub action to automate branch creation and synchronization.
- [ ] Test the GitHub action to ensure that `main` is created as an exact copy of `master`.
- [ ] Verify that the `main` branch is updated automatically after every merge into `master`.
- [ ] Document the changes and update any relevant README or contributing guidelines.

### PR DESCRIPTION
**Pull request by nollymar**
Refs: #29840 

### Proposed Changes
* Branch `main` is created if not exists
* `main` is updated each time a PR gets merged into master

This PR fixes: #29840

### CODE REVIEW

File: .github/workflows/cicd_post_sync-main-with-master.yml
Comment (dcolina) [inline .github/workflows/cicd_post_sync-main-with-master.yml:17]: The checkout version should be `v4`.

Comment (nollymar) [reply .github/workflows/cicd_post_sync-main-with-master.yml]: Thank you Dani! Fixed!

**Review by dcolina — APPROVED** [review]

**Comment by spbolton** [comment]
Do we need to be careful here with branch protections and on commits into the new main branch.  In the short term only this script should be able to push to main

File: .github/workflows/cicd_post_sync-main-with-master.yml
Comment (spbolton) [reply .github/workflows/cicd_post_sync-main-with-master.yml:33]: Hmm,  I agree but just thinking does it use those values for the push,  note we are not committing anything here so all the commits should move over to the new branch as is.  But maybe it is required for github push

Comment (nollymar) [reply .github/workflows/cicd_post_sync-main-with-master.yml:38]: @dcolina I have pushed a change to include the git configuration.
@spbolton I set the same branch protections that master has

Comment (spbolton) [reply .github/workflows/cicd_post_sync-main-with-master.yml:38]: @nollymar Ultimately it should have the same, the problem is that the default prevents push directly to the branch without going through PR/merge , unless the machine user is allowed to bypass this then the push will not work.   
e.g. "Require a pull request before merging is enabled".    As it is temporary then we may be ok to allow the machine user, but we have to be prepared that not allowing direct pushes is a security measure and think about what an attacker can do if getting hold of the machine user token to bypass other security checks.   Be aware that adding the machine token at the repository level allows anyone able to create a PR on the repository (all our developers only, or if an attacker gets a token or password for a developer) could adjust the PR workflow or create a new one (but this is restricted by our codeowners validation) to directly access the token value and then bypass push using it.     In this case I think we are ok adding the bypass while we are migrating over.

**Review by valentinogiardino — APPROVED** [review]

**Pull request merged**
